src/twisted/web/resource.py

```diff
@@ -240,7 +243,7 @@ def render(self, request):
 
         @see: L{IResource.render}
         """
-        m = getattr(self, 'render_' + nativeString(request.method), None)
+        m = getattr(self, u'render_' + request.method.decode("ascii"), None)
```

Comment:
`getattr` works with unicode strings on Python 2 and Python 3.  I confirmed by trying it on both.

Reply:
On Python 2, `getattr` will silently encode unicode strings with the ASCII codec.  So this will decode `request.method` as ASCII, concatenate it with u`render_`, then re-encode it as ASCII.

On Python 3, prefixing a string literal with `u` has no effect, and `request.method.decode('ascii')` is equivalent to `nativeString`.

So, yes it works, but `getattr`'s actual API works in terms of native strings. The original code aligned with that API by concatenating a native string with `nativeString`.